### PR TITLE
[mobile][photos] add dismiss button to crafting memories banner

### PR DESCRIPTION
**Pull request by aspizu**
Simulator Screenshot - iPhone 17 Pro - 2026-08-03 at 15 44 31
Simulator Screenshot - iPhone 17 Pro - 2026-08-03 at 15 44 46

### CODE REVIEW

File: mobile/apps/photos/lib/settings/local_settings.dart [hunk 1]
```diff
@@ -698,4 +701,12 @@ class LocalSettings {
       await _prefs.setString(_kDeletePreference, preference._serializedValue);
     }
   }
+
+  Future<void> setCraftingMemoriesBannerDismissed() async {
+    await _prefs.setBool(_kCraftingMemoriesBannerDismissed, true);
```
[the remark this answers is not in the document]
Comment (aspizu) [reply mobile/apps/photos/lib/settings/local_settings.dart:706]: shutup

File: mobile/apps/photos/lib/ui/home/memories/craft_memories.dart [hunk 2]
```diff
@@ -44,102 +46,143 @@ class _CraftMemoriesState extends State<CraftMemories> {
...
+                  message: l10n.close,
+                  child: GestureDetector(
+                    behavior: HitTestBehavior.opaque,
+                    onTap: () async {
+                      await localSettings.setCraftingMemoriesBannerDismissed();
+                      if (!mounted) return;
+                      widget.onNotificationsPermissionGranted?.call();
+                    },
+                    child: SizedBox.square(
+                      dimension: 20,
```
Comment (aspizu) [inline mobile/apps/photos/lib/ui/home/memories/craft_memories.dart]: **MEDIUM** — The close control's whole tap target is only 20×20 pixels. A near miss lands on the banner behind it and opens the notification permission/settings flow instead of dismissing, so this is hard to use and worse for users with limited dexterity. Keep the 20 px icon but wrap it in a 44–48 px transparent tap target (or use an IconButton), and add a widget test that taps near the icon edge and confirms dismissal runs without requesting permission.

**Review by aspizu — COMMENTED** [review]
Automated code review: high- and medium-severity findings only.
